caput/pfb.py

```diff
@@ -0,0 +1,190 @@
+"""Tools for calculating the effects of the CASPER tools PFB.
+
+This module can:
+- Evaluate the typical window functions used
+- Evaluate a python model of the PFB
+- Calculate the decorrelation effect for signals offset by a known time delay.
+
+Window functions
+================
+
+.. autosummary::
+    :toctree: generated/
+
+    sinc_window
+    sinc_hanning
+    sinc_hamming
+
+PFB
+===
+
+.. autosummary::
+    :toctree: generated/
+
+    pfb
+    decorrelation_ratio
+"""
+# === Start Python 2/3 compatibility
+from __future__ import absolute_import, division, print_function, unicode_literals
+from future.builtins import *  # noqa  pylint: disable=W0401, W0614
+from future.builtins.disabled import *  # noqa  pylint: disable=W0401, W0614
+
+# === End Python 2/3 compatibility
+
+import numpy as np
+from scipy.interpolate import interp1d
+
+
+def sinc_window(ntap, lblock):
+    """Sinc window function.
+
+    Parameters
+    ----------
+    ntaps : integer
+        Number of taps.
+    lblock: integer
+        Length of block.
+
+    Returns
+    -------
+    window : np.ndarray[ntaps * lblock]
+    """
+    # Sampling locations of sinc function
+    X = np.linspace(-ntap / 2, ntap / 2, ntap * lblock, endpoint=False)
+
+    # np.sinc function is sin(pi*x)/pi*x, not sin(x)/x, so we can just X
+    return np.sinc(X)
+
+
+def sinc_hanning(ntap, lblock):
+    """Hanning-sinc window function.
+
+    Parameters
+    ----------
+    ntaps : integer
+        Number of taps.
+    lblock: integer
+        Length of block.
+
+    Returns
+    -------
+    window : np.ndarray[ntaps * lblock]
+    """
+
+    return sinc_window(ntap, lblock) * np.hanning(ntap * lblock)
+
+
+def sinc_hamming(ntap, lblock):
+    """Hamming-sinc window function.
+
+    Parameters
+    ----------
+    ntaps : integer
+        Number of taps.
+    lblock: integer
+        Length of block.
+
+    Returns
+    -------
+    window : np.ndarray[ntaps * lblock]
+    """
+
+    return sinc_window(ntap, lblock) * np.hamming(ntap * lblock)
+
+
+def pfb(timestream, nfreq, ntap=4, window=sinc_hamming):
```

Comment:
I would rename this chime_pfb - pub is too general for a chime-specific code.

Reply:
Oh, I think this is just an issue with the doc-string... this isn't chime-specific at all.